server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/meta/StarTreeMetaWriter.java

```diff
@@ -0,0 +1,156 @@
+     * @param dataFileLength         the length of the star tree data file
+     * @throws IOException if an I/O error occurs while serializing the metadata
+     */
+    public static void writeStarTreeMetadata(
+        IndexOutput metaOut,
+        StarTreeField starTreeField,
+        List<MetricAggregatorInfo> metricAggregatorInfos,
+        Integer segmentAggregatedCount,
+        long dataFilePointer,
+        long dataFileLength
+    ) throws IOException {
+
+        long initialMetaFilePointer = metaOut.getFilePointer();
+
+        writeMetaHeader(metaOut, CompositeMappedFieldType.CompositeFieldType.STAR_TREE, starTreeField.getName());
+        writeMeta(metaOut, metricAggregatorInfos, starTreeField, segmentAggregatedCount, dataFilePointer, dataFileLength);
+
+        logger.debug(
+            "Star tree meta size in bytes : {} for star-tree field {}",
+            metaOut.getFilePointer() - initialMetaFilePointer,
+            starTreeField.getName()
+        );
+    }
+
+    /**
+     * Writes the star-tree metadata header.
+     *
+     * @param metaOut            the IndexOutput to write the header
+     * @param compositeFieldType the composite field type of the star-tree field
+     * @param starTreeFieldName  the name of the star-tree field
+     * @throws IOException if an I/O error occurs while writing the header
+     */
+    private static void writeMetaHeader(
+        IndexOutput metaOut,
+        CompositeMappedFieldType.CompositeFieldType compositeFieldType,
+        String starTreeFieldName
+    ) throws IOException {
+        // magic marker for sanity
+        metaOut.writeLong(COMPOSITE_FIELD_MARKER);
+
+        // version
+        metaOut.writeVInt(VERSION_CURRENT);
+
+        // star tree field name
+        metaOut.writeString(starTreeFieldName);
+
+        // star tree field type
+        metaOut.writeString(compositeFieldType.getName());
+    }
+
+    /**
+     * Writes the star-tree metadata.
+     *
+     * @param metaOut                   the IndexOutput to write the metadata
+     * @param metricAggregatorInfos     the list of metric aggregator information
+     * @param starTreeField             the star tree field
+     * @param segmentAggregatedDocCount the aggregated document count for the segment
+     * @param dataFilePointer           the file pointer to the start of the star-tree data
+     * @param dataFileLength            the length of the star-tree data file
+     * @throws IOException if an I/O error occurs while writing the metadata
+     */
+    private static void writeMeta(
+        IndexOutput metaOut,
+        List<MetricAggregatorInfo> metricAggregatorInfos,
+        StarTreeField starTreeField,
+        Integer segmentAggregatedDocCount,
+        long dataFilePointer,
+        long dataFileLength
+    ) throws IOException {
+
+        // number of dimensions
+        metaOut.writeVInt(starTreeField.getDimensionsOrder().size());
```

Comment:
Not related to this PR itself.

Lets add a simple todo to check for number of dimensions here to handle upcoming changes.

Reply:
let's still keep in the scope of this PR. Those things can come with timestamp PRs

Reply:
just add a todo, am not asking for any changes :D so that we don't miss it. I found it really hard to resolve merges